src/jobs/config_sync_job.ts

```diff
@@ -377,6 +489,133 @@ export class ConfigSyncJob extends SyncJob {
     // if any of those operations did fail, return false
     return operationsOk;
   }
+
+  /*
+  checkes if we sync timer-tiket or ticket-ticket tools
+  returns true if ticket-ticket, false othewise
+  */
+  private _isTicket2TicketSync() {
+    return (this._connection.firstService.name === 'Jira' || this._connection.firstService.name === 'Redmine')
+      ? (this._connection.secondService.name === 'Jira' || this._connection.secondService.name === 'Redmine')
+      : false
+  }
+
+  private async _createTicket2TicketIssueMappings(firstServiceObjects: ServiceObject[], secondServiceObjects: ServiceObject[]): Promise<Mapping[]> {
+    const firstServiceIssuesWithCustField = firstServiceObjects.filter((o: ServiceObject) => {
+      return o.syncCustomFieldValue
+    })
+    const secondServiceIssuesWithCustField = secondServiceObjects.filter((o: ServiceObject) => {
+      return o.syncCustomFieldValue
+    })
+    const newMappings: Mapping[] = new Array()
+    const firstSecond = await this._createTicket2TicketMapping(firstServiceObjects, secondServiceIssuesWithCustField, true)
+    const secondFirst = await this._createTicket2TicketMapping(secondServiceObjects, firstServiceIssuesWithCustField, false)
+    newMappings.push(...firstSecond)
+    newMappings.push(...secondFirst)
+    return newMappings
+  }
+
+  private async _createTicket2TicketMapping(firstServiceObjects: ServiceObject[], secondServiceObjectsWithCustField: ServiceObject[], first: boolean): Promise<Mapping[]> {
+    const newMappings: Mapping[] = new Array()
+    for (const secondObject of secondServiceObjectsWithCustField) {
+      const idOfFirst = await getIdOfAnotherServiceIdFromLink(first ? this._connection.firstService : this._connection.secondService, secondObject.syncCustomFieldValue)
+      if (idOfFirst) {//'second objects has link to the first => first is primary
+        const firstObject = firstServiceObjects.find((o: ServiceObject) => {
+          return o.id == idOfFirst
+        })
+        if (firstObject) {
+          //check if projects corespond with those that should be paired
+          const areInTheProjectPair = this._connection.projectMappings.filter((p: ProjectMapping) => {
+            (p.idFirstService === firstObject.projectId && p.idSecondService === secondObject.projectId)
+              || (p.idFirstService === secondObject.projectId && p.idSecondService === firstObject.projectId)
```

Comment:
Co je idFirstService? Nějak mi to pojmenování nesedí vzhledem k porovnávané hodnotě.

Reply:
Asi dáme call a vyřešíme to tam

Reply:
Honza se pokusí upravit, aby bylo výstižnější.

Reply:
Přejmenováno